src/js/components/stock-movement-wizard/request/AddItemsPage.jsx

```diff
@@ -806,10 +806,21 @@ class AddItemsPage extends Component {
             demandPerReplenishmentPeriod,
             quantityOnHand,
             quantityAllowed,
+            version,
           } = val;
 
-          let qtyRequested = 0;
-          if (quantityRequested) qtyRequested = quantityRequested;
+          // Determine the quantity requested values to set for each row. If the request already has
+          // a quantity value set, use that, otherwise calculate a value to use. When creating an
+          // e-request, if QoH for an item > the quantity in the stocklist item, then the quantity
+          // will have been set to 0 initially. For that scenario, we want to try to recompute
+          // quantity here (in the add items step) for pull type replenishments so that the quantity
+          // factors in demand, but we only want to do this if the row hasn't been manually adjusted
+          // to 0 (if a user sets their own quantity requested, that should persist). We distinguish
+          // that case by checking if the item version is 0, meaning it hasn't been modified.
+          // This isn't a perfect solution but is the least intrusive one for now. Probably a better
+          // option would be to move this logic to the stock movement create logic on the backend.
```

Comment:
I'm scared to do it but I feel like we need to move this logic to the backend. I can't think of another solution... It's way too close to release for me to want to actually do that though